azurerm/internal/services/network/vpn_gateway_data_source.go

```diff
@@ -0,0 +1,253 @@
+						"peer_weight": {
+							Type:     pluginsdk.TypeInt,
+							Computed: true,
+						},
+
+						"bgp_peering_address": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"instance_0_bgp_peering_address": {
+							Type:     pluginsdk.TypeList,
+							Computed: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"custom_ips": {
+										Type:     pluginsdk.TypeSet,
+										Computed: true,
+										Elem: &pluginsdk.Schema{
+											Type: pluginsdk.TypeString,
+										},
+									},
+
+									"ip_configuration_id": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+
+									"default_ips": {
+										Type:     pluginsdk.TypeSet,
+										Computed: true,
+										Elem: &pluginsdk.Schema{
+											Type: pluginsdk.TypeString,
+										},
+									},
+
+									"tunnel_ips": {
+										Type:     pluginsdk.TypeSet,
+										Computed: true,
+										Elem: &pluginsdk.Schema{
+											Type: pluginsdk.TypeString,
+										},
+									},
+								},
+							},
+						},
+
+						"instance_1_bgp_peering_address": {
+							Type:     pluginsdk.TypeList,
+							Computed: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"custom_ips": {
+										Type:     pluginsdk.TypeSet,
+										Computed: true,
+										Elem: &pluginsdk.Schema{
+											Type: pluginsdk.TypeString,
+										},
+									},
+
+									"ip_configuration_id": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+
+									"default_ips": {
+										Type:     pluginsdk.TypeSet,
+										Computed: true,
+										Elem: &pluginsdk.Schema{
+											Type: pluginsdk.TypeString,
+										},
+									},
+
+									"tunnel_ips": {
+										Type:     pluginsdk.TypeSet,
+										Computed: true,
+										Elem: &pluginsdk.Schema{
+											Type: pluginsdk.TypeString,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"scale_unit": {
+				Type:     pluginsdk.TypeInt,
+				Computed: true,
+			},
+
+			"tags": tags.SchemaDataSource(),
+		},
+	}
+}
+
+func dataSourceVPNGatewayRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.VpnGatewaysClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] VPN Gateway %q was not found in Resource Group %q - removing from state", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving VPN Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if resp.ID != nil {
+		d.SetId(*resp.ID)
+	}
```

Comment:
rather than relying on the Azure ID, can we add a Resource ID Parser for this and use that value instead? (see this PR for an example https://github.com/terraform-providers/terraform-provider-azurerm/pull/10128)